pkg/dashboard/store/metrics/metrics.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package metrics
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/fx"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/dashboard/core"
+)
+
+const chaosDashboardSubsystem = "chaos_dashboard"
+
+// Collector implements prometheus.Collector interface
+type Collector struct {
+	experimentStore core.ExperimentStore
+	scheduleStore   core.ScheduleStore
+	workflowStore   core.WorkflowStore
+
+	archivedExperiments *prometheus.GaugeVec
+	archivedSchedules   *prometheus.GaugeVec
+	archivedWorkflows   *prometheus.GaugeVec
+}
+
+// NewCollector initializes metrics and collector
+func NewCollector(experimentStore core.ExperimentStore, scheduleStore core.ScheduleStore, workflowStore core.WorkflowStore) *Collector {
+	return &Collector{
+		experimentStore: experimentStore,
+		scheduleStore:   scheduleStore,
+		workflowStore:   workflowStore,
+		archivedExperiments: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Subsystem: chaosDashboardSubsystem,
+			Name:      "archived_experiments",
+			Help:      "Total number of archived chaos experiments",
+		}, []string{"namespace", "type"}),
+		archivedSchedules: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Subsystem: chaosDashboardSubsystem,
+			Name:      "archived_schedules",
+			Help:      "Total number of archived chaos schedules",
+		}, []string{"namespace"}),
+		archivedWorkflows: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Subsystem: chaosDashboardSubsystem,
+			Name:      "archived_workflows",
+			Help:      "Total number of archived chaos workflows",
+		}, []string{"namespace"}),
+	}
+}
+
+// Describe implements the prometheus.Collector interface.
+func (collector *Collector) Describe(ch chan<- *prometheus.Desc) {
+	collector.archivedExperiments.Describe(ch)
+	collector.archivedSchedules.Describe(ch)
+	collector.archivedWorkflows.Describe(ch)
+}
+
+// Collect implements the prometheus.Collector interface.
+func (collector *Collector) Collect(ch chan<- prometheus.Metric) {
+	collector.collectArchivedExperiments()
+	collector.collectArchivedSchedules()
+	collector.collectArchivedWorkflows()
+	collector.archivedExperiments.Collect(ch)
+	collector.archivedSchedules.Collect(ch)
+	collector.archivedWorkflows.Collect(ch)
+}
+
+func (collector *Collector) collectArchivedExperiments() {
```

Comment:
How often does this function run? This function will query all the data in the store. Is it better to collect the metric when the experiment is archived?

Reply:
> How often does this function run?
It depends on how often the Prometheus server pull/scrape/collect data from `chaos-dashboard`. Usually the most frequently collect pattern is every 15 seconds.


> Is it better to collect the metric when the experiment is archived?

I prefer to count all the records in the DB at each time. If we calculate/count the number of archived experiments when we archive them, there needs more logic to resolve about things like `chaos-dashboard` process restart and the user manually updates/ deletes the records in the database.

Reply:
I have considered the resource usage for reading all the records in the table from the database, I think it is just OK, for reading thousands of records from sqlite and MySQL every 15 seconds. 

But using `count` instead of `list-all` in store-layer/SQL is necessary, I think we should make this improvement in the next PRs.

Reply:
OK, We can optimize it at the SQL level first

Reply:
need to create a issue for this